app/src/main/java/com/portfolio/tasky/views/TaskyAppCompatEditText.kt

```diff
@@ -1,62 +1,117 @@
+class TaskyAppCompatEditText(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs), OnTouchListener   {
+
+    lateinit var subLayout: LayoutInputFieldBinding
     private var error : Boolean = false
+    private var isPassword : Boolean = false
     private var drawableLast : Drawable? = null
     private lateinit var editText : AppCompatEditText
     private var inputType : Int = 0
     private var hint : String? = null
 
+
     init {
         setupView(attrs)
-        isError()
+        setError(false)
         setDrawableLast()
         setInputType()
         setHint()
+        setValid(false)
+        isPassword()
+        setTouchListener()
     }
 
     private fun setHint() {
-        viewBinding.hint = hint
+        subLayout.hint = hint
     }
 
     private fun setInputType() {
-        viewBinding.etInput.inputType = inputType
+        subLayout.etInput.inputType = inputType
     }
 
     private fun setDrawableLast() {
-        viewBinding.drawableLast = drawableLast
+        subLayout.drawableLast = drawableLast
+    }
+
+    fun setError(error : Boolean){
+        subLayout.error = error
     }
 
-    fun isError(){
-        viewBinding.error = error
+    fun setValid(valid: Boolean){
+        subLayout.valid = valid
+    }
+    private fun isPassword(){
+        subLayout.password = isPassword
+    }
+
+    private fun setTouchListener() {
+        subLayout.etInput.setOnTouchListener(this)
     }
 
     @SuppressLint("CustomViewStyleable")
     private fun setupView(attrs: AttributeSet) {
         val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
-        viewBinding = LayoutInputFieldBinding.inflate(inflater, this, false)
-        addView(viewBinding.root)
+        subLayout = LayoutInputFieldBinding.inflate(inflater, this, false)
+        addView(subLayout.root)
 
-        editText = viewBinding.etInput
+        editText = subLayout.etInput
         val typedArray: TypedArray = context.obtainStyledAttributes(attrs, R.styleable.EditTextInputField)
         error = typedArray.getBoolean(R.styleable.EditTextInputField_error, false)
         drawableLast = typedArray.getDrawable(R.styleable.EditTextInputField_drawableLast)
         inputType = typedArray.getInt(R.styleable.EditTextInputField_inputType, InputType.TYPE_CLASS_TEXT)
         hint = typedArray.getString(R.styleable.EditTextInputField_hint)
+        isPassword = typedArray.getBoolean(R.styleable.EditTextInputField_isPassword, false)
         typedArray.recycle()
     }
+
+    override fun onTouch(v: View, event: MotionEvent): Boolean {
+        when (event.action) {
+            MotionEvent.ACTION_UP -> {
+                showHidePassword(event)
+            }
+        }
+        return false
+    }
+
+    private fun showHidePassword(event: MotionEvent) {
+        val textLocation = IntArray(2)
+        if (event.rawX >= textLocation[0] + subLayout.etInput.width - subLayout.etInput.totalPaddingRight) {
```

Comment:
What does this if statement do?

I recommend creating a variable that describes what it's doing, and then checking that variable

Example:
```
val isValidPositionForInput = event.rawX >= textLocation[0] + subLayout.etInput.width - subLayout.etInput.totalPaddingRight
```

This is just an example, you'd name it based on what it does

Reply:
It actually makes the end area of AppCompatEditText clickable.